RELEASE.md

```diff
@@ -1,33 +1,65 @@
-To release a new version of qtpy on PyPI:
+# Release Procedure
 
-* Close Github milestone
+To release a new version of QtPy on PyPI:
 
-* git fetch upstream && git merge upstream/master
 
-* git clean -xfdi
+## Prepare
 
-* Update CHANGELOG.md using `loghub` to generate the list of issues and PRs merged to add at the top of the file
+* Close Github milestone and ensure all issues are resolved/moved
 
-    loghub -m vX.X.X spyder-ide/qtpy
+* `git switch <release-branch>` then `hub sync` or `git fetch upstream && git merge upstream/<release-branch>`
```

Comment:
```suggestion
* `git fetch upstream && git merge upstream/master`
```

* Instructions should be as simple as possible (just copy/paste), so no need to add two methods here and specially one that requires another tool besides git.
* `1.x` is an exceptional case of a release branch. Most probably in the future we won't maintain others.

Reply:
Sure, fair enough on both. I wasn't sure about including both to begin with, and it actually makes it sufficiently ambigous that even you were unclear that `git switch master` was supposed to apply to both. Before I go ahead with this, should we keep the other couple lines that offers two methods (`pip` and `conda`) as they are, or should we remove the `pip` alternative and only keep `conda` there too?

Reply:
Also in the spirit of simplification, is there a reason you specify `git fetch upstream && git merge upstream/master`, when `git pull upstream master` is equivalent? Per the [docs](http://git-scm.com/docs/git-pull),

> In its default mode, `git pull` is shorthand for `git fetch` followed by `git merge FETCH_HEAD`.

(Since `FETCH_HEAD` in the case of the given `pull` command is `upstream/master`)

Reply:
I went ahead and simplified this line to just `git switch master && git pull upstream master`; lmk what you want me to do about the `pip` and `conda` alternatives

Reply:
> lmk what you want me to do about the pip and conda alternatives

What do you mean by this?

Reply:
Oh, I see. Yes, we prefer to use `pip` to do releases because the build and upload dependencies are always up to date.

So please remove the commands related to conda and leave a mention saying why we prefer to use pip.

Reply:
Sorry, I missed seeing this reply somehow.

Makes sense; I removed the conda command and the separate install step (since `pip install --upgrade` does both).

I also changed the `pip` invocation to use `python -m pip` to be cross-platform compatible, moved the `git clean` step to right before the build, where it belongs (and to balance out the sections a bit), and added an explict `pip install --upgrade loghub` command to install/upgrade loghub.

Reply:
Also, made the Pytest invocation in the release checklist more consistent and robust, and (in a separate commit) added `pip check` to the release checklist, and `pip check` and `twine check` to the CIs.